app/models/mission.rb

```diff
@@ -1,3 +1,74 @@
 class Mission < ActiveRecord::Base
-  # Remember to create a migration!
+  has_many :rest_points
+  has_many :passengers
+  has_many :dispatches
+  has_many :trailer_dispatches
+
+  def self.create_new(params)
+
+    self.transaction do
+
+      mission = self.new
+
+      mission.step_off      ="#{params[:mission][:step_off_date]+" "+params[:mission][:step_off_time]+":00"}"
+      mission.return        ="#{params[:mission][:return_date]+" "+params[:mission][:return_time]+":00"}"
+      mission.name          = params[:mission][:name]
+      mission.unit_serviced = params[:mission][:unit_serviced]
+
+      mission.dispatches << self.set_up_dispatches(params[:trucks])
+      mission.passengers << self.set_up_passengers(params[:passengers])
+
+          # this is also another method; needs refactoring
+      if params[:trailers]
+        trailer_dispatches = []
+        params[:trailers].each do |trailer|
+          trailer = Trailer.find_by!(name: trailer[:trailer_name])
+          trailer_dispatches << TrailerDispatch.create!(trailer: trailer)
+        end
+        mission.trailer_dispatches = trailer_dispatches
+      end
+
+      if params[:payloads]
+        params[:payloads].each do |payload|
+          if /^TL.*/.match(payload[:vehicle_name])
+            Payload.create!(
+              payload: payload[:payload],
+              vehicle_type: "Trailer",
+              vehicle_id: Trailer.find_by!(name: payload[:vehicle_name]).id)
+          else
+            Payload.create!(
+              payload: payload[:payload],
+              vehicle_type: "Truck",
+              vehicle_id: Truck.find_by!(name: payload[:vehicle_name]).id)
+          end
+        end
+      end
+      mission.save!
+    end
+  end
+
+  def self.set_up_dispatches(trucks)
```

Comment:
should a mission itself have the responsibility to set up the dispatches and passengers? i'm not sure what makes the most sense, but i just wanted to throw that out there!